lib/saml/url_service.rb

```diff
@@ -122,8 +123,9 @@ def url_settings
     end
 
     def add_query(url, params)
```

Comment:
I would expect `add_query('https://foo.com/bar?baz=a',{foo: 'b'})` to return `https://foo.com/bar?baz=a&foo='b` not `https://foo.com/bar?foo=b` , but I really like that you're using `URI.parse`